tests/TestHierarchy.hs

```diff
@@ -0,0 +1,26 @@
+module TestHierarchy where
+
+import System.Process (readCreateProcess, shell)
+
+import Test.Hspec (Spec, describe, hspec, it, shouldBe)
+
+import TestUtils (pushd)
+
+main :: IO ()
+main = pushd "tests/support/hierarchy" (hspec spec)
+
+spec :: Spec
+spec = do
+  describe "hierarchy" $ do
+    it "generates usable graphviz graphs" $ do
+      let expected = unlines
+            [ "digraph Main {"
+            , "  A;"
+            , "  A -> B;"
+            , "  C;"
+            , "}"
+            ]
+
+      actual <- readCreateProcess (shell "purs hierarchy src/Main.purs") ""
```

Comment:
Having spent far too long messing about with CI scripts, my general approach is the fewer shells, the better. I think they're a nightmare. I reckon it's very likely that a test that relies on them will break because of a different value of the PATH variable in some environment somewhere, or something like that.

All the other tests in this repo work the same way: we don't test the CLI for the other commands, just the library part. But after moving as much as is reasonable into the library part, I think that's fine.